sale_order_line_multi_warehouse/models/sale_order_line.py

```diff
@@ -0,0 +1,266 @@
+# Copyright 2024 Manuel Regidor <manuel.regidor@sygel.es>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    sale_order_line_warehouse_ids = fields.One2many(
+        string="SO Line Warehouses",
+        comodel_name="sale.order.line.warehouse",
+        inverse_name="order_line_id",
+    )
+    qty_assigned_to_warehouse = fields.Float(
```

Comment:
Did you thought making the product_uom_qty computed instead of creating this field, and using the compute + compute_reverse function instead of overriding create and write functions?